great_expectations/data_context/types/base.py

```diff
@@ -1484,21 +1488,26 @@ def update(
                 self.expectation_suite_name = other_config.expectation_suite_name
             # update
             if other_config.batch_request is not None:
-                batch_request = (
-                    self.batch_request.get_json_dict()
-                    if isinstance(self.batch_request, BatchRequest)
-                    else self.batch_request
-                )
-                batch_request = batch_request or {}
-                other_batch_request = (
-                    other_config.batch_request.get_json_dict()
-                    if isinstance(other_config.batch_request, BatchRequest)
-                    else other_config.batch_request
+                if self.batch_request is not None:
+                    batch_data = self.batch_request.batch_data
+                    batch_request = self.batch_request.get_json_dict()
+                    batch_request["batch_data"] = batch_data
+                else:
+                    batch_request = {}
+
+                other_config_batch_data = other_config.batch_request.batch_data
+                other_batch_request = other_config.batch_request.get_json_dict()
+                other_batch_request["batch_data"] = other_config_batch_data
+
+                updated_batch_request = nested_update(
+                    batch_request, other_batch_request
                 )
-                batch_request.update(other_batch_request)
-                self.batch_request = batch_request
+                self.batch_request = BatchRequest(**updated_batch_request)
             if other_config.action_list is not None:
-                self.update_action_list(other_action_list=other_config.action_list)
+                self.action_list = self.get_updated_action_list(
+                    base_action_list=self.action_list,
+                    other_action_list=other_config.action_list,
+                )
```

Comment:
Some of this code seems to be same or very similar to that in checkpoint.py -- if appropriate, is there an opportunity for reuse?..

Reply:
It's already re-used, but the other way around. I made `get_updated_action_list` a classmethod, which gets used in Checkpoint